tests/pytest_tests/system_tests/test_launch/test_launch_vertex.py

```diff
@@ -0,0 +1,124 @@
+
+
+@pytest.fixture
+def mock_vertex_environment():
+    """Mock an instance of the GcpEnvironment class."""
+    environment = MagicMock()
+    environment.region.return_value = "europe-west-4"
+
+
+def test_vertex_resolved_submitted_job(relay_server, monkeypatch):
+    def mock_launch_vertex_job(*args, **kwargs):
+        return args[1]
+
+    mock_env = MagicMock(spec=GcpEnvironment)
+    monkeypatch.setattr(
+        wandb.sdk.launch.loader,
+        "environment_from_config",
+        lambda *args: mock_env,
+    )
+    monkeypatch.setattr(
+        wandb.sdk.launch.loader,
+        "builder_from_config",
+        lambda *args: MagicMock(),
+    )
+    monkeypatch.setattr(
+        wandb.sdk.launch.loader,
+        "registry_from_config",
+        lambda *args: None,
+    )
+
+    monkeypatch.setattr(
+        "wandb.sdk.launch.runner.local_container.docker_image_exists",
+        lambda x: None,
+    )
+    monkeypatch.setattr(
+        "wandb.sdk.launch.runner.local_container.pull_docker_image",
+        lambda x: None,
+    )
+    monkeypatch.setattr(
+        "wandb.sdk.launch.builder.noop.NoOpBuilder.build_image",
+        lambda *args, **kwargs: "testimage",
+    )
+
+    monkeypatch.setattr(
+        "wandb.sdk.launch.runner.vertex_runner.launch_vertex_job",
+        mock_launch_vertex_job,
+    )
+
+    with relay_server():
+        entity_name = "test_entity"
+        project_name = "test_project"
+        entry_command = ["python", "test.py"]
+
+        # test with user provided image
+        project = MagicMock()
+        entrypoint = EntryPoint("blah", entry_command)
+        project.resource_args = {
+            "vertex": {
+                "run": {"restart_job_on_worker_restart": False},
+                "spec": {
+                    "staging_bucket": "gs://test_bucket",
+                    "worker_pool_specs": [
+                        {
+                            "machine_spec": {
+                                "machine_type": "n1-highmem-4",
+                                "accelerator_type": "NVIDIA_TESLA_T4",
+                                "accelerator_count": 1,
+                            },
+                            "replica_count": 1,
+                            "container_spec": {"image_uri": "${image_uri}"},
+                        }
+                    ],
+                },
+            }
+        }
+        project.fill_macros.return_value = project.resource_args
+        project.target_entity = entity_name
+        project.target_project = project_name
+        project.name = None
+        project.run_id = "asdasd"
+        project.sweep_id = "sweeeeep"
+        project.override_config = {}
+        project.override_entrypoint = entrypoint
+        project.get_single_entry_point.return_value = entrypoint
+        project.override_args = ["--a1", "20", "--a2", "10"]
+        project.docker_image = "testimage"
+        project.image_name = "testimage"
+        project.job = "testjob"
```

Comment:
```suggestion
        project.job = "testjob"
        project.queue_name = None
        project.queue_entity = None
        project.run_queue_item_id = None
```

Reply:
I think because of a change made last week we need to set these attributes to `None`, or else we get unexpected environment variables in the test.

Reply:
https://app.circleci.com/pipelines/github/wandb/wandb/27881/workflows/0010ca81-041c-4d3c-b48e-6175b009a3b9/jobs/845433/tests

Reply:
@Hojland I think we still need to commit this suggested change in order your system test to pass

Reply:
I'm sorry! Thought I did